skcosmo/metrics/_reconstruction_measures.py

```diff
@@ -0,0 +1,651 @@
+
+import numpy as np
+
+from joblib import Parallel, delayed
+
+
+def pointwise_global_reconstruction_error(
+    X,
+    Y,
+    train_idx=None,
+    test_idx=None,
+    scaler=None,
+    estimator=None,
+):
+    """Computes the pointwise global reconstruction error using the source X
+    to reconstruct the features or samples of target Y based on a minimization
+    by linear regression:
+
+    .. math:: GRE^{(i)}(X,Y) = \min_W ||y_i - x_iW|| % # noqa: W605
+
+    If used with X and Y of shape (n_samples, n_features) it computes the pointwise
+    global reconstruction error of the features as defined in Ref. [frm]_.
+    In this case the number of samples of X and Y should agree with each other,
+    but the number of features can be different. The error is expressed per sample.
+
+    If used with X and Y of shape(n_features, n_samples) it computes the
+    reconstruction error of the samples. In this case the number of
+    features of X and Y should agree with each other, but the number of
+    samples can be different. The error is expressed per feature.
+
+    The default parameters mimics the ones of Ref. [frm]_.
+
+    Parameters
+    ----------
+    X : array_like of shape (n_samples, X_n_features)
+        Source data which reconstructs target Y.
+        For feature reconstruction of Y using X use input shape (samples, features).
+        For sample reconstruction of Y using X use input shape (features, samples).
+
+    Y : array_like of shape (n_samples, Y_n_targets)
+        Target data which is reconstructed with X.
+        For feature reconstruction of Y using X use input shape (samples, features).
+        For sample reconstruction of Y using X use input shape (features, samples).
+
+    train_idx : array_like, default=None
+        array of indices used for training, if None,
+        If None, the complement of the ``test_idx`` is used. If ``train_size`` is
+        also None, 2-fold split is taken.
+
+    test_idx : array_like, default=None
+        array of indices used for training, if None,
+        If None, the complement of the ``train_idx`` is used. If ``test_size`` is
+        also None, 2-fold split is taken.
+
+    scaler : object implementing fit/transfom
+        Scales the X and Y before computing the reconstruction measure.
+        The default value scales the features such that the reconstruction
+        measure on the training set is upper bounded to 1.
+
+    estimator : object implementing fit/predict, default=None
+        Sklearn estimator used to reconstruct features/samples.
+
+    Returns
+    -------
+    pointwise_global_reconstruction_error : array_like
+        The global reconstruction error for each sample/point
+
+    References
+    ----------
+    .. [frm] Goscinski, Fraux, Imbalzano and Ceriotti. "The role of feature space in
+             atomistic learning." arXiv preprint arXiv:2009.02741 (2020).
+    """
+    (
+        train_idx,
+        test_idx,
+        scaler,
```

Comment:
Is it possible to pass a null scaler?

Reply:
Please answer this question.

Reply:
So the input of all reconstruction measures look more or less like this
```
def reconstruction_measure(
    X,
    Y,
    train_idx=None,
    test_idx=None,
    scaler=None,
    estimator=None,
):
```
If it is None, then the default scaler is used. If one does not want to use a scaler, then one needs to give a dummy scaler as input, which just returns the identity